### PR TITLE
driver: lpspi: add the lpspi driver for rt1170

### PR DESCRIPTION
**Pull request by CristXu**
enable the  lpspi1 for the rt1170 evk board

Signed-off-by: Crist Xu <crist.xu@nxp.com>

### CODE REVIEW

**Review by MaureenHelm — APPROVED** [review]

**Review by mmahadevan108 — APPROVED** [review]

**Comment by MaureenHelm** [comment]
Please rebase

**Comment by galak** [comment]
@CristXu you may need my CI fix here as well - https://github.com/zephyrproject-rtos/zephyr/pull/36887

**Review by galak — APPROVED** [review]

File: samples/drivers/jesd216/sample.yaml [hunk 1]
```diff
@@ -4,6 +4,7 @@ tests:
   sample.drivers.jesd216:
     tags: spi flash
     filter: dt_compat_enabled("jedec,spi-nor")
+    platform_exclude: mimxrt1170_evk_cm7 mimxrt1170_evk_cm4
```
Comment (galak) [inline samples/drivers/jesd216/sample.yaml:7]: Hmm, is there a reason we need this.  Can we not enable CONFIG_MEMC_MCUX_FLEXSPI for rt1170?

Comment (CristXu) [reply samples/drivers/jesd216/sample.yaml:7]: hi， 
    in fact, the CONFIG_MEMC_MCUX_FLEXSPI is not enabled, so there is a link error, said  **undefined reference**. and also the rt1170's flash is with the "jedec,spi-nor" compatible: **compatible = "issi,is25wp128", "jedec,spi-nor";** so this test will run for the rt1170, but without include the SPI driver(which in rt1170 is flexspi, not spi instead). so i think that we
can use the platform_exclude to filter the rt1170_evk board.


Regards,
   Crist

**Comment by CristXu** [comment]
related this PR: #36887

**Pull request merged**